tests/robot-tests/scripts/create_snapshots.py

```diff
@@ -45,27 +46,28 @@ def __init__(self, public_url: str, slack_webhook_url: str):
 
     def _gets_parsed_html_from_page(self, url: str) -> BeautifulSoup:
         response = self.session.get(url, stream=True)
-        assert response.status_code == 200, f"Requests response wasn't 200!\nResponse: {response}"
+        assert response.status_code == 200, f"Failed to get page: {url}. Status code: {response.status_code}"
         return BeautifulSoup(response.text, "html.parser")
 
     def _get_create_snapshot_func(self, name: str):
-        if name == "data_catalogue_snapshot.json":
-            return self._create_data_catalogue_snapshot
-        elif name == "find_statistics_snapshot.json":
-            return self._create_find_statistics_snapshot
-        elif name == "methodologies_snapshot.json":
-            return self._create_methodologies_snapshot
-        elif name == "table_tool_snapshot.json":
-            return self._create_table_tool_snapshot
-        else:
-            raise Exception(f"Invalid snapshot name: {name}")
+        switcher = {
+            "data_catalogue_snapshot.json": self._create_data_catalogue_snapshot,
+            "find_statistics_snapshot.json": self._create_find_statistics_snapshot,
+            "methodologies_snapshot.json": self._create_methodologies_snapshot,
+            "table_tool_snapshot.json": self._create_table_tool_snapshot,
+        }
+
+        return switcher.get(name, lambda: "Invalid snapshot name")
```

Comment:
I think that this returns a string rather than a function could be confusing when trying to debug this, if someone tries to access a snapshot function that doesn't exist.

I think I'd prefer for this to fail immediately if that happens. Maybe:
```
result = switcher.get(name)
if result is None:
  print(f"waaah! no such snapshot function for '{name}'!")
  os.exit(1)
return result
```